astropy/units/core.py

```diff
@@ -1512,30 +1512,19 @@ def to_system(self, system):
         Returns
         -------
         units : list of `CompositeUnit`
-            The list is ranked so that units containing only the base
-            units of that system will appear first.
+            The list is that returned by `~astropy.units.BaseUnit.compose`,
+            but sorted by the fewest number of non-base units, so that
+            preference is given to, e.g., "cm / s^2" over "Gal" (in cgs).
+
         """
         bases = set(system.bases)
 
         def score(compose):
-            # In case that compose._bases has no elements we return
-            # 'np.inf' as 'score value'.  It does not really matter which
-            # number we would return. This case occurs for instance for
-            # dimensionless quantities:
-            compose_bases = compose.bases
-            if len(compose_bases) == 0:
-                return np.inf
-            else:
-                sum = 0
-                for base in compose_bases:
-                    if base in bases:
-                        sum += 1
-
-                return sum / float(len(compose_bases))
+            return len(set(compose.bases) - bases)
 
         x = self.decompose(bases=bases)
         composed = x.compose(units=system)
-        composed = sorted(composed, key=score, reverse=True)
```

Comment:
Hmm why was it reversed in the first place?

Reply:
Previously, it wanted the highest possible number of base units, so reversing made sense. Now it wants the lowest possible number of non-base units (but in the system).